### PR TITLE
Add ROBOTOLOGY_PROJECT_TAGS to specify which tags/branches to use for each CMake package

### PR DESCRIPTION
**Pull request by traversaro**
Add `ROBOTOLOGY_PROJECT_TAGS` to specify which tags/branches to use for each CMake package. 
The problem in switching between branches when the superbuild is already configured are
properly documented in the `change-project-tags.md` document.

The main purpose of this change is to improve usability for users of the superbuild in the IIT's labs in Genoa, that typically use the unstable branches of the robotology projects.

Furthermore, it also addresses some other issues: 
* It adds the `ROBOTOLOGY_PROJECT_TAGS_CUSTOM_FILE` option, to specify externally the version to use, hopefully addressing some of the concerns discussed in https://github.com/robotology/robotology-superbuild/issues/227 
* It starts using the `yarp-3.2` branch of YARP as YARP stable branch, as requested in https://github.com/robotology/robotology-superbuild/issues/259 . Furthermore, as not all the robotology repos will be using the `master` branch as stable branch and the `devel` as unstable branch, renames the `ProjectTags***.cmake` files from `Master`/`Devel` to `Stable`/`Unstable` @pattacini 
* It simplifies testing in CI the `Unstable` branches, as requested by https://github.com/robotology/robotology-superbuild/issues/101, ~and  done in https://github.com/robotology/robotology-superbuild/pull/275/commits/8e26c4126ed7fca568f0775eeac8bd23ff70b769~  The CI part is blocked by https://github.com/robotology/wearables/issues/57, so it will be moved to a future PR.

Hopefully, this should fix https://github.com/robotology/robotology-superbuild/issues/33 .

### CODE REVIEW

**Comment by traversaro** [comment]
@valegagge @vtikha I added Valentina as a reviewer because this can be related to your work, feel free to decline the invitation!

**Comment by traversaro** [comment]
@kouroshD @Yeshasvitvs as profile mantainers I added you as reviewers so you can be aware of this changes, and  review the specific branches/tags used in the ProjectTags***.cmake files for the packages of your profiles.

File: doc/change-project-tags.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+`ROBOTOLOGY_PROJECT_TAGS` is an advanced CMake option, so it is visible from the CMake GUI only after enabling the visualization of advanced variables.
+It can take one of three possible values: 
+* `Stable` :  this is the default value, that will use the tags or branches specified in [`cmake/ProjectTagsStable.cmake`](cmake/ProjectTagsStable.cmake) 
+              that are the stables branches for the robotology projects.
+* `Unstable` : by selecting this option, you will use the tags or branches specified in [`cmake/ProjectTagsUnstable.cmake`](cmake/ProjectTagsUnstable.cmake), 
+               that are the "unstable" active development branches for the robotology projects. This is reccomended just for users that work at the IIT labs in Genoa,
+               as it can break compilation or runtime behaviour without any notice.
+* `Custom` : by selecting this option, you need to manually specify the tags or branches that you want to use by specifying a custom project tags file in the 
+             `ROBOTOLOGY_PROJECT_TAGS_CUSTOM_FILE` CMake option. This is useful when  you want to use a fixed version of the software built by the superbuild.
```
Comment (diegoferigo) [inline doc/change-project-tags.md]: If I understood the logic, I would also write that this file can be either kept outside the superbuild source directory by specifying an absolute path, or relative and without the `.txt` extension if stored in the `cmake/` folder (and maybe commit it in a temporarily fork / branch).

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,11 +83,21 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
...
-# Optionally load a file named ProjectsTagsMaster
-# for overwriting the desired branch/tag/commit
-# of the included subprojects of the superbuild
-# Useful for having superbuild "branches"
-include(ProjectsTagsMaster OPTIONAL)
+set(ROBOTOLOGY_PROJECT_TAGS "Stable" CACHE STRING "The tags to be used for the robotology projects: Stable, Unstable or Custom. This can be changed only before the first configuration.")
+mark_as_advanced(ROBOTOLOGY_PROJECT_TAGS)
+set(ROBOTOLOGY_PROJECT_TAGS_CUSTOM_FILE CACHE FILEPATH "If ROBOTOLOGY_PROJECT_TAGS is custom, this file will be loaded to specify the tags of the projects to use.")
+mark_as_advanced(ROBOTOLOGY_PROJECT_TAGS_CUSTOM_FILE)
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Stable" "Unstable" "Custom")
```
Comment (nunoguedelha) [inline CMakeLists.txt]: shouldn't it be `ROBOTOLOGY_PROJECT_TAGS` instead od `CMAKE_BUILD_TYPE` here?

Comment (traversaro) [reply CMakeLists.txt]: Good catch! This was a copy paste error.

**Comment by traversaro** [comment]
@diegoferigo @nunoguedelha I should have address your comments.

**Comment by traversaro** [comment]
cc @gabrielenava

**Pull request merged**